### ISSUE TITLE
Make custom code Drupal 9 compatible

### ISSUE THREAD

**Issue by markfullmer** [writecrow/crow_backend#20]
## Description of task
Probable subtasks:
1. Run drupal-check to find deprecated code
2. Add core version requirements to all custom modules

**Comment by markfullmer** [comment]
Output of automated deprecation checking (not too bad):

```
 ------ -------------------------------------------------------------------------- 
  Line   corpus_importer/src/DeDupeHelper.php                                      
 ------ -------------------------------------------------------------------------- 
  37     Call to deprecated method entityManager() of class Drupal:                
         in drupal:8.0.0 and is removed from drupal:9.0.0.                         
         Use \Drupal::entityTypeManager() instead in most cases. If the needed     
         method is not on \Drupal\Core\Entity\EntityTypeManagerInterface, see the  
         deprecated \Drupal\Core\Entity\EntityManager to find the                  
         correct interface or service.                                             
 ------ -------------------------------------------------------------------------- 

 ------ -------------------------------------------------------------------------- 
  Line   corpus_importer/src/ImporterHelper.php                                    
 ------ -------------------------------------------------------------------------- 
  263    Call to deprecated method entityManager() of class Drupal:                
         in drupal:8.0.0 and is removed from drupal:9.0.0.                         
         Use \Drupal::entityTypeManager() instead in most cases. If the needed     
         method is not on \Drupal\Core\Entity\EntityTypeManagerInterface, see the  
         deprecated \Drupal\Core\Entity\EntityManager to find the                  
         correct interface or service.                                             
  292    Call to deprecated method entityManager() of class Drupal:                
         in drupal:8.0.0 and is removed from drupal:9.0.0.                         
         Use \Drupal::entityTypeManager() instead in most cases. If the needed     
         method is not on \Drupal\Core\Entity\EntityTypeManagerInterface, see the  
         deprecated \Drupal\Core\Entity\EntityManager to find the                  
         correct interface or service.                                             
 ------ -------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------- 
  Line   corpus_importer/src/ImporterService.php                                          
 ------ --------------------------------------------------------------------------------- 
  81     Call to deprecated function drupal_set_message():                                
         in drupal:8.5.0 and is removed from drupal:9.0.0.                                
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.             
  416    Call to deprecated constant FILE_CREATE_DIRECTORY: Deprecated in drupal:8.7.0    
         and is removed from drupal:9.0.0. Use                                            
         \Drupal\Core\File\FileSystemInterface::CREATE_DIRECTORY.                         
  416    Call to deprecated function file_prepare_directory():                            
         in drupal:8.7.0 and is removed from drupal:9.0.0.                                
         Use \Drupal\Core\File\FileSystemInterface::prepareDirectory().                   
  417    Call to deprecated constant FILE_EXISTS_REPLACE: Deprecated in drupal:8.7.0 and  
         is removed from drupal:9.0.0. Use                                                
         \Drupal\Core\File\FileSystemInterface::EXISTS_REPLACE.                           
  432    Call to deprecated function drupal_set_message():                                
         in drupal:8.5.0 and is removed from drupal:9.0.0.                                
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.             
  435    Call to deprecated function drupal_set_message():                                
         in drupal:8.5.0 and is removed from drupal:9.0.0.                                
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.             
  438    Call to deprecated function drupal_set_message():                                
         in drupal:8.5.0 and is removed from drupal:9.0.0.                                
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.             
 ------ --------------------------------------------------------------------------------- 

 ------ --------------------------------------------------------------------------------- 
  Line   custom/content_type_resource/content_type_resource.module                        
 ------ --------------------------------------------------------------------------------- 
  27     Call to deprecated constant FILE_CREATE_DIRECTORY: Deprecated in drupal:8.7.0    
         and is removed from drupal:9.0.0. Use                                            
         \Drupal\Core\File\FileSystemInterface::CREATE_DIRECTORY.                         
  27     Call to deprecated function file_prepare_directory():                            
         in drupal:8.7.0 and is removed from drupal:9.0.0.                                
         Use \Drupal\Core\File\FileSystemInterface::prepareDirectory().                   
  28     Call to deprecated constant FILE_EXISTS_REPLACE: Deprecated in drupal:8.7.0 and  
         is removed from drupal:9.0.0. Use                                                
         \Drupal\Core\File\FileSystemInterface::EXISTS_REPLACE.                           
 ------ --------------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------- 
  Line   custom/corpus_search/src/Controller/CorpusSearch.php                         
 ------ ----------------------------------------------------------------------------- 
  53     Call to deprecated constant REQUEST_TIME: Deprecated in drupal:8.3.0 and is  
         removed from drupal:9.0.0. Use \Drupal::time()->getRequestTime();            
 ------ ----------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------- 
  Line   custom/corpus_search/src/SearchService.php                                   
 ------ ----------------------------------------------------------------------------- 
  29     Call to deprecated function db_like():                                       
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,                   
         get a database connection injected into your service from the container      
         and call escapeLike() on it. For example,                                    
  37     Call to deprecated function db_like():                                       
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,                   
         get a database connection injected into your service from the container      
         and call escapeLike() on it. For example,                                    
  44     Call to deprecated function db_like():                                       
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,                   
         get a database connection injected into your service from the container      
         and call escapeLike() on it. For example,                                    
  47     Call to deprecated function db_like():                                       
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,                   
         get a database connection injected into your service from the container      
         and call escapeLike() on it. For example,                                    
  58     Call to deprecated constant REQUEST_TIME: Deprecated in drupal:8.3.0 and is  
         removed from drupal:9.0.0. Use \Drupal::time()->getRequestTime();            
  191    Call to deprecated constant REQUEST_TIME: Deprecated in drupal:8.3.0 and is  
         removed from drupal:9.0.0. Use \Drupal::time()->getRequestTime();            
 ------ ----------------------------------------------------------------------------- 

 ------ ----------------------------------------------------------------------------- 
  Line   custom/corpus_search/src/TextMetadata.php                                    
 ------ ----------------------------------------------------------------------------- 
  60     Call to deprecated constant REQUEST_TIME: Deprecated in drupal:8.3.0 and is  
         removed from drupal:9.0.0. Use \Drupal::time()->getRequestTime();            
 ------ ----------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------- 
  Line   word_frequency/src/FrequencyService.php                                  
 ------ ------------------------------------------------------------------------- 
  40     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  50     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  53     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  98     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  108    Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  111    Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  210    Call to deprecated function drupal_set_message():                        
         in drupal:8.5.0 and is removed from drupal:9.0.0.                        
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.     
  322    Call to deprecated function drupal_set_message():                        
         in drupal:8.5.0 and is removed from drupal:9.0.0.                        
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.     
  325    Call to deprecated function drupal_set_message():                        
         in drupal:8.5.0 and is removed from drupal:9.0.0.                        
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead.     
  328    Call to deprecated function drupal_set_message():                        
         in drupal:8.5.0 and is removed from drupal:9.0.0.                        
         Use \Drupal\Core\Messenger\MessengerInterface::addMessage() instead. 
```

**Comment by markfullmer** [comment]
Remaining deprecations that will need to be manually fixed after the commit of #23 :+1: 

```
 ------ ------------------------------------------------------------------------------ 
  Line   corpus/corpus.install                                                         
 ------ ------------------------------------------------------------------------------ 
  30     Call to deprecated constant USER_REGISTER_ADMINISTRATORS_ONLY: Deprecated in  
         drupal:8.3.0 and is removed from drupal:9.0.0. Use                            
         \Drupal\user\UserInterface::REGISTER_ADMINISTRATORS_ONLY instead.             
 ------ ------------------------------------------------------------------------------ 

 ------ ------------------------------------------------------------------------- 
  Line   corpus/modules/custom/corpus_search/src/SearchService.php                
 ------ ------------------------------------------------------------------------- 
  29     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  37     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  44     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  47     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
 ------ ------------------------------------------------------------------------- 

 ------ ------------------------------------------------------------------------- 
  Line   corpus/modules/word_frequency/src/FrequencyService.php                   
 ------ ------------------------------------------------------------------------- 
  40     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  50     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  53     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  98     Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  108    Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example,                                
  111    Call to deprecated function db_like():                                   
         in drupal:8.0.0 and is removed from drupal:9.0.0. Instead,               
         get a database connection injected into your service from the container  
         and call escapeLike() on it. For example, 
```

**Comment by markfullmer** [comment]
This has been resolved through #20 and elsewhere.
